data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/RecordDispatcherImpl.java

```diff
@@ -192,6 +196,23 @@ private void onDeadLetterSinkFailure(final KafkaConsumerRecord<Object, CloudEven
     finalProm.complete();
   }
 
+  private static KafkaConsumerRecord<Object, CloudEvent> buildCloudEventValueIfMissing(KafkaConsumerRecord<Object, CloudEvent> record) {
+    // A valid CloudEvent in the CE binary protocol binding of Kafka
+    // might be composed by only Headers.
+    //
+    // KafkaConsumer doesn't call the deserializer if the value
+    // is null.
+    //
+    // That means that we get a record with a null value event though
+    // the record is a valid CloudEvent.
+    if (record.value() == null) {
+      logDebug("Value is null", record);
+      final var value = cloudEventDeserializer.deserialize(record.record().topic(), record.record().headers(), null);
+      return new KafkaConsumerRecordImpl<>(KafkaConsumerRecordUtils.copyRecordAssigningValue(record.record(), value));
+    }
+    return record;
```

Comment:
`buildCloudEventValueIfMissing`

aren't we here "building" (deserializing) the value of the `KafkaConsumerRecord`, that happens to be a `CloudEvent`? 


Could we change to `if (record.value() != null) { return record; } else { ....}` ? Not sure, but that might be easier to read 


`deserialize(record.record().topic(), record.record().headers(), null);` I understand why we pass in null, but.... where is else is the deserialization done?  why is it here again invoked, just b/c it is null :man_shrugging:

Reply:
> Could we change to if (record.value() != null) { return record; } else { ....} ? Not sure, but that might be easier to read

I agree!

> deserialize(record.record().topic(), record.record().headers(), null); I understand why we pass in null, but.... where is else is the deserialization done? why is it here again invoked, just b/c it is null 

The deserialization is done inside the KafkaConsumer, but when `value == null`, it doesn't call the deserializer since there is nothing to deserialize, so we need to deserialize the record manually when this happens since only record headers constitute a valid CloudEvent.

Reply:
>  but when `value == null`, it doesn't call the deserializer

so that's OK for pure/vanilla kafka - but is that a problem for other "protocols" on top? Or is that just a corner-case here :man_shrugging: 

>  we need to deserialize the record manually when this happens since only record headers constitute a valid CloudEvent.

I just wonder how to make this more obvious - I understand we do it... perhaps the `buildCloudEventValueIfMissing` name is also not perfect ? :see_no_evil:   Sorry, just trying to make it "simpler"

Reply:
> Sorry, just trying to make it "simpler"

Nah, we're doing code review for this reason, right? :heart_eyes: 

I changed the name to `maybeDeserializeValueFromHeaders` and changed the `if .. ` as suggested.